delfin/drivers/dell_emc/power_store/rest_handler.py

```diff
@@ -0,0 +1,958 @@
+        for ip_address in ip_pool_address:
+            purposes_list = ip_address.get('purposes')
+            if consts.MGMT_NODE_COREOS not in purposes_list:
+                continue
+            address = ip_address.get('address')
+            appliance_id = ip_address.get('appliance_id')
+            node_id = ip_address.get('node_id')
+            ip_d['{}{}'.format(appliance_id, node_id)] = address
+        return ip_d
+
+    def get_port_hardware(self):
+        hardware_d = {}
+        hardware_list = self.rest_call(self.REST_HARDWARE_URL)
+        for hardware in hardware_list:
+            hardware_d[hardware.get('id')] = hardware
+        return hardware_d
+
+    def get_fc_ports(self, storage_id, hardware_d):
+        list_fc_ports = []
+        fc_res = self.rest_call(self.REST_FC_PORT_URL)
+        for fc in fc_res:
+            appliance_id = fc.get('appliance_id')
+            name = fc.get('name')
+            is_link_up = fc.get('is_link_up')
+            connection_status = consts.PORT_CONNECTION_STATUS_MAP.get(
+                is_link_up, constants.PortConnectionStatus.UNKNOWN)
+            lifecycle_state = hardware_d.get(
+                fc.get('sfp_id'), {}).get('lifecycle_state')
+            health_status = consts.PORT_HEALTH_STATUS_MAP.get(
+                lifecycle_state, constants.PortHealthStatus.UNKNOWN)
+            fc_port_result = {
+                'name': name,
+                'storage_id': storage_id,
+                'native_port_id': fc.get('id'),
+                'location': '{}:{}'.format(appliance_id, name),
+                'connection_status': connection_status,
+                'health_status': health_status,
+                'type': constants.PortType.FC,
+                'speed': self.convert_speed(fc.get('current_speed')),
+                'max_speed': self.convert_speed(fc.get('supported_speeds')),
+                'native_parent_id': fc.get('node_id'),
+                'wwn': fc.get('wwn')
+            }
+            list_fc_ports.append(fc_port_result)
+        return list_fc_ports
+
+    @staticmethod
+    def convert_speed(supported_speeds):
+        max_speed = None
+        if supported_speeds:
+            if isinstance(supported_speeds, list):
+                supported_speed = supported_speeds[
+                    consts.DigitalConstant.MINUS_ONE_INT]
+            else:
+                supported_speed = supported_speeds
+            if '_Gbps' in supported_speed:
+                supported_speed = supported_speed.replace('_Gbps', '')
+                max_speed = int(supported_speed) * units.G
+            elif '_Mbps' in supported_speed:
+                supported_speed = supported_speed.replace('_Mbps', '')
+                max_speed = int(supported_speed) * units.M
+            elif '_Kbps' in supported_speed:
+                supported_speed = supported_speed.replace('_Kbps', '')
+                max_speed = int(supported_speed) * units.k
+        return max_speed
```

Comment:
Please use `return` statement instead of using `if ... elif ... elif ...` , like:
```
if not supported_speeds:
    return
supported_speed = supported_speeds[consts.DigitalConstant.MINUS_ONE_INT] if isinstance(supported_speeds, list) else supported_speeds
if '_Gbps' in supported_speed:
    supported_speed = supported_speed.replace('_Gbps', '')
    return int(supported_speed) * units.G
if ...
```

Reply:
It has been modified.
@staticmethod
    def convert_speed(supported_speeds):
        if not supported_speeds:
            return
        supported_speed = \
            supported_speeds[consts.DigitalConstant.MINUS_ONE_INT]\
            if isinstance(supported_speeds, list) else supported_speeds
        if '_Gbps' in supported_speed:
            supported_speed = supported_speed.replace('_Gbps', '')
            return int(supported_speed) * units.G
        if '_Mbps' in supported_speed:
            supported_speed = supported_speed.replace('_Mbps', '')
            return int(supported_speed) * units.M
        if '_Kbps' in supported_speed:
            supported_speed = supported_speed.replace('_Kbps', '')
            return int(supported_speed) * units.k